lib/vsc/mympirun/pmi/slurm.py

```diff
@@ -0,0 +1,246 @@
+#
+# Copyright 2019-2019 Ghent University
+#
+# This file is part of vsc-mympirun,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# the Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/hpcugent/vsc-mympirun
+#
+# vsc-mympirun is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# vsc-mympirun is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with vsc-mympirun.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+Slurm PMI class, i.e. wrap around srun
+"""
+
+import os
+import re
+
+from vsc.mympirun.pmi.sched import Sched
+from vsc.mympirun.pmi.pmi import PMI, PMIX
+from vsc.utils.run import async_to_stdout
+
+class Slurm(Sched):
+
+    _sched_for = ['slurm']
+    SCHED_ENVIRON_ID = 'SLURM_JOB_ID'
+
+    LAUNCHER = 'srun'
+
+    def run_function(self):
+        """
+        srun itself supports redirection to file
+        """
+        args = []
+        if self.options.output:
+            args = ['--output=%s' % self.options.output]
+
+        return async_to_stdout, args
+
+    def pmicmd_sched(self):
+        """Generate the sched related arguments to the launcher as a list"""
+        args = []
+
+        # sbatch will set the sbatch parameters as environment variables and they will get picked up by srun
+        args.append("--chdir=" + os.getcwd())
```

Comment:
where would i hardcode it? (i'll remove the comment)

Reply:
My question is more why is this hardcoded at all? You should clarify this with a comment...

It's probably because `srun` runs from home directory by default?

Reply:
the comment used to clarify it. i'll put it back